packages/wallet/backend/src/grant/service.ts

```diff
@@ -80,6 +80,32 @@ export class GrantService implements IGrantService {
       args.filter = { identifier: { in: identifiers } }
     }
 
-    return await this.rafikiAuthService.listGrantsWithPagination(args)
+    const grants = await this.rafikiAuthService.listGrantsWithPagination(args)
+    this.parseIntervals(grants.grants.edges.map((it) => it.node))
+    return grants
+  }
+
+  private parseIntervals(grants: Grant[]) {
+    for (const grant of grants)
+      for (const accessElement of grant.access)
+        if (accessElement.limits?.interval) {
+          const time = accessElement.limits?.interval.split('/')
+          const duration = moment.duration(time[2])
+          const date = moment(time[1])
+          const months = duration.asMonths()
+          if (months === 1)
+            accessElement.limits!.interval = `every month from ${date.format('ddd')}`
+          else if (months > 1 && months < 12)
+            accessElement.limits!.interval = `every ${months} month from ${date.format('ddd')}`
+          else if (months === 12)
+            accessElement.limits!.interval = `every year from ${date.format('ddd')}`
+          else if (months > 12) {
```

Comment:
the implementation looks to be focused on well defined intervals.
for the interval `R5/2008-03-01T13:00:00Z/P1Y2M10DT2H30M`
it will be displayed that it repeats every year, right? that will give the user incorrect info